### PR TITLE
feature: new useLeafFocusedNode

### PR DESCRIPTION
**Pull request by LBRDan**
This works using focusedNodeId from focusStore and passing it down to useFocusNode hook

Answers to #76 

For details follow previous PR #80 

Also, adds something in Docs (Readme) and basic tests (heavily inspired by useFocusNode ones)

### CODE REVIEW

File: src/hooks/use-leaf-focused-node.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    const unsubscribe = contextValue.store.subscribe(checkForSync);
+
+    return () => {
+      unsubscribe();
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+  const focusNode = useFocusNode(focusNodeId);
+  return focusNodeId ? focusNode : null;
+}
```
[the remark this answers is not in the document]
Comment (jamesplease) [reply src/hooks/use-leaf-focused-node.ts:56]: > Is there any chance that store.focusedNodeId and focusHierarchy[focusHierarchy.length - 1]?.focusId may be different at a given time?

They should always match! (unless there's a bug 😁)

[the remark this answers is not in the document]
Comment (LBRDan) [reply src/hooks/use-leaf-focused-node.ts:56]: Okay then, I'll update this PR according your amazing solution!

But before this, I'll make another PR with the empty string check inside `focus-node.ts` and treat as non valid

**Comment by jamesplease** [comment]
Thanks so much! Will try to get this released this week.

**Comment by jamesplease** [comment]
Released as `0.0.18`. Thanks a ton @LBRDan !

**Pull request merged**